### DISCUSSION TITLE
Remove header/side menu

### DISCUSSION THREAD

**Discussion by mbrevda** [music-assistant/support#2666] [Q&A] [answered]
When embedding in a lovelace view using in ingress path, how can the header/side menu be hidden? My use case it kiosk mode where I have the menus hidden (and certainly don't need the same menu twice).

<img width="455" alt="Screenshot 2024-07-23 at 6 21 10 PM" src="https://github.com/user-attachments/assets/3433df0d-1ab0-467c-b5fc-36a89afaf2e8">

**Comment by OzGav** [comment]
I believe you will have to expose and use port 8095

**Reply by mbrevda** [reply]
That is it's own can of worms as I access my dashboard over ssl, which the stand alone port doesn't support, making it impossible to load via an iframe.

**Reply by OzGav** [reply]
Then I don't think it is possible.

**Reply by mbrevda** [reply]
I see. Would it be feasible to allow injection of css to MA? This would allow for the menu to be hidden via css. This isn't possible at the moment because existing methods (such as card-mod) cannot access the iframe

**Reply by OzGav** [reply, marked as answer]
Most things are possible however dev time is short. I would expect this would be overtaken by a custom card however as that is on the roadmap.

**Reply by mbrevda** [reply]
I've worked around this by proxying the web port through nginx by adding the following to `/config/nginx_proxy_default_music.conf`: 

```
location /_x_music/ {
    proxy_pass http://homeassistant.local.hass.io:8095/;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

     # WebSocket support
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
}
```

**Reply by rononline** [reply]
@mbrevda but can you then inject css?? How did you that??

**Reply by mbrevda** [reply]
I didnt inject css. I use (I think) wallpanel which hides the side bar on the wall displays. Using the above, I created a new frontend view, and added MA like:

```yaml
type: iframe
url: /_x_music/#/home
aspect_ratio: 50%
```

if you wanted to, you might be able to inject CSS, too, with something like this (please use google for more info on this):

```
location /proxied-path/ {
    proxy_pass http://backend-server;
    sub_filter '<head>' '<head><style>body { background-color: lightblue; }</style>';
    sub_filter_once off;
    sub_filter_types text/html;
}
